src/vision/status_quo/barbara_anguishes_over_http.md

```diff
@@ -0,0 +1,47 @@
+# 😱 Status quo stories: Barbara Anguishes Over HTTP
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect people's experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+Barbara is starting a new project, working together with Alan. They want to write a Rust library and as part of it they will need to make a few HTTP calls to various web services. While HTTP is part of the responsibilities of the library it is by no means the only thing the library will need to do.
+
+As they are pair programming, they get the part of the library where HTTP will be involved and Alan asks Barbara, "OK, how do I make an HTTP request?".
+
+As an experienced async Rust developer Barbara has been dreading this question from the start of the project. She's tempted to ask "How long do you have?", but she quickly gathers herself and sstarts to outline the various considerations. She starts with a relatively simple question: "Should we use an HTTP library with a sync interface or an async interface?".
+
+Alan, somewhat confused by the question since this isn't a concern in the languages he's used to, replies with his own question: "I don't know, what do you think?". Barbara thinks for a second. She knows that bridging sync and async can be difficult, and so there's another question they need to answer first: "Well, are we going to expose a sync or an async interface to the users of our library?".
```

Comment:
> somewhat confused by the question since this isn't a concern in the languages he's used to, replies with his own question

Is it really not? C# has `async` functions, right, how does this divide get bridged?

Reply:
Also, what language is Alan used to? Let's specify

Reply:
C# behaves like Rust I believe in that it's not a good idea to mix async and sync code. But many languages don't have async at all. In the real world scenario, the person I was talking to has experience in a wide range of languages and wasn't too thrown by the need to not mix async/sync, but I've heard from many people this isn't something they are used to.

Reply:
If Alan came from JS, say, there'd only be one choice -- and it would be async. So it might be that we could say Alan comes from there and this also explains a bit more to why he picked "async" as the obvious choice.

Reply:
As I read this story, I view the "Alan is somewhat confused" part as just a tool to get Alan and Barbara to have this out-loud back-and-forth dialog.  So I don't think he necessarily needs to be confused, perhaps he just needs to be "unsure" enough to throw the question back to Barbara.

But to give Alan a plausible reason to think that async is the way to go, perhaps he is indeed coming from the JS world where he remembers the pain when the JS/npm ecosystem slowly made the transition from a callbacks to the "async" and "await" keywords.   Mixing a library with callbacks and a library with async is a little annoying and perhaps Alan thought of the JS world "It sure would be nice is everything was async" (and thus applies similar thinking here)

Reply:
> I guess the difficulty is when you go from red to blue, back to red — because you can't make an executor inside an executor.

From https://fasterthanli.me/articles/pin-and-suffering, I have experienced the same thing. Using sync code within async is okay, async code within sync code is okay, but when you do async code within sync code within async code, you will likely run into an issue since the cannot have two runtimes together. Usually in rust when it compiles it works but in this case the compiler is not able to detect this behavior beforehand so it will be an error in the runtime.